it/controllers/pensions/annualAllowance/PensionProviderTaxPaidAnnualAllowanceControllerISpec.scala

```diff
@@ -0,0 +1,488 @@
+  )
+
+  private implicit val url: Int => String = pensionProviderTaxPaidAnnualAllowanceUrl
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      import Selectors._
+      import user.commonExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "render did your pension schemes pay or agree to pay the tax with no pre-filling" which {
+          lazy val result: WSResponse = showPage(user, anPensionsUserDataEmptyCya)
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, checked = Some(false))
+          radioButtonCheck(noText, 2, checked = Some(false))
+          buttonCheck(buttonText, continueButtonSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          textOnPageCheck(totalNonUkTax, amountText)
+          inputFieldValueCheck(amountInputName, amountInputSelector, "")
+          formPostLinkCheck(pensionProviderTaxPaidAnnualAllowanceUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render page with 'yes' pre-filled and amount field set" which {
+
+          val pensionAnnualAllowanceViewModel = aPensionAnnualAllowanceViewModel.copy(
+            pensionProvidePaidAnnualAllowanceQuestion = Some(true),
+            taxPaidByPensionProvider = Some(BigDecimal(existingAmount)))
+          val pensionUserData = pensionsUserDataWithAnnualAllowances(pensionAnnualAllowanceViewModel, isPriorSubmission = false)
+
+          lazy val result: WSResponse = showPage(user, pensionUserData)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, checked = Some(true))
+          radioButtonCheck(noText, 2, checked = Some(false))
+          buttonCheck(buttonText, continueButtonSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          textOnPageCheck(totalNonUkTax, amountText)
+          inputFieldValueCheck(amountInputName, amountInputSelector, existingAmount)
+          formPostLinkCheck(pensionProviderTaxPaidAnnualAllowanceUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render page with 'no' pre-filled and amount field set to 0" which {
```

Comment:
Shouldn't it be " and no amount set"?

Reply:
Yes, wording updated to reflect latest change to None

Reply:
Confirmed.